### PR TITLE
Full title match is better than split match

### PR DESCRIPTION
**Pull request by leonardr**
This branch adds two new ways a book might show up in search results: its title or one of its authors might be a near-exact match against the search term.

This should resolve the 'modern romance' and 'law of the mountain man' problems described in https://github.com/NYPL-Simplified/server_core/issues/482, where a book whose title includes a genre loses out to books that are in that genre. It may also resolve some of the 'game of thrones' problems described in that issue.

This branch changes the Elasticsearch configuration used by core during Travis runs so that everything runs in a single shard; this eliminates the possibility that search result ordering will be unpredictable because of how documents are assigned to shards.

This branch also changes the way Elasticsearch is configured: instead of separate configuration settings for the alias and the index, a single configuration setting controls the prefix used for both the alias and the index. This means that changing the alias to point to some other index is currently an unsupported operation -- it's supported by the code (transfer_current_alias) but nobody ever calls that method.

This is a good time to make that configuration change because this branch introduces a v3 of the search document, which requires that the entire index be rebuilt anyway. (The migration script should take care of this.)

### CODE REVIEW

File: external_search.py [hunk 1]
```diff
@@ -453,6 +453,13 @@ def make_target_age_query(target_age):
         match_phrase = make_phrase_query(query_string, ['title.minimal', 'author', 'series.minimal'])
         must_match_options.append(match_phrase)
 
+        # An exact title or author match outweighs a match that is split
+        # across fields.
+        match_title = make_phrase_query(query_string, ['title.minimal'], 150)
```
Comment (aslagle) [inline external_search.py]: What happened to `title.standard`?

**Comment by aslagle** [comment]
I think you need a migration for this - for the configuration setting change, if nothing else.

**Comment by leonardr** [comment]
I forgot to add the migration script. All it does is create a brand new index using the default prefix. I don't think there are any instances where this won't work, and if there are any, we don't know enough about their setup to automatically migrate them.

**Comment by leonardr** [comment]
I could also add a migration script to _remove_ the old configuration settings, just to avoid any confusion when looking at the database in the future.

File: tests/test_external_search.py [hunk 2]
```diff
@@ -778,6 +836,143 @@ def query(*args, **kwargs):
...
+            title="Gumby In Love",
+            authors="Pokey",
+            audience=Classifier.AUDIENCE_YOUNG_ADULT, genre="Romance"
+        )
+        self.ya_romance.presentation_edition.subtitle = (
+            "Modern Fairytale Series, Book 3"
+        )
+
+        # TODO: Uncomment these lines and the 'modern romance'
+        # test fails for some reason.
```
Comment (aslagle) [inline tests/test_external_search.py]: you fixed this!

File: external_search.py [hunk 3]
```diff
@@ -751,6 +753,64 @@ def map_fields(cls, fields, field_description):
...
+                "type": "string",
+                "analyzer": "en_analyzer",
+                "fields": {
+                    "minimal": {
+                        "type": "string",
+                        "analyzer": "en_minimal_analyzer"},
+                    "standard": {
+                        "type": "string",
+                        "analyzer": "standard"
+                    }
```
Comment (leonardr) [inline external_search.py]: This is the part that's different.

**Pull request merged**